lib/steep/services/hover_provider/ruby.rb

```diff
@@ -65,22 +65,22 @@ def project
         def method_definition_for(factory, type_name, singleton_method: nil, instance_method: nil)
           case
           when instance_method
-            factory.definition_builder.build_instance(type_name).methods[instance_method]
+            factory.definition_builder.build_instance(type_name).methods.fetch(instance_method)
           when singleton_method
             methods = factory.definition_builder.build_singleton(type_name).methods
 
             if singleton_method == :new
-              methods[:new] || methods[:initialize]
+              methods[:new] || methods.fetch(:initialize)
```

Comment:
This is also a bit strange. The left hand is `#[]`, but the right hand is `#fetch`. It's a little difficult to understand the behavior for me.